Library/Homebrew/test/spec_helper.rb

```diff
@@ -273,39 +276,7 @@
       Requirement.clear_cache
       Readall.clear_cache if defined?(Readall)
 
-      FileUtils.rm_rf [
-        *TEST_DIRECTORIES,
-        *Keg::MUST_EXIST_SUBDIRECTORIES,
-        HOMEBREW_LINKED_KEGS,
-        HOMEBREW_PINNED_KEGS,
-        HOMEBREW_PREFIX/"var",
-        HOMEBREW_PREFIX/"Caskroom",
-        HOMEBREW_PREFIX/"Frameworks",
-        HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-cask",
-        HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-bar",
-        HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-bundle",
-        HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-foo",
-        HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-services",
-        HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-shallow",
-        HOMEBREW_LIBRARY/"PinnedTaps",
-        HOMEBREW_REPOSITORY/".git",
-        CoreTap.instance.path/".git",
-        CoreTap.instance.alias_dir,
-        CoreTap.instance.path/"formula_renames.json",
-        CoreTap.instance.path/"tap_migrations.json",
-        CoreTap.instance.path/"audit_exceptions",
-        CoreTap.instance.path/"style_exceptions",
-        CoreTap.instance.path/"pypi_formula_mappings.json",
-        *Pathname.glob("#{HOMEBREW_CELLAR}/*/"),
-      ]
-
-      files_after_test = Test::Helper::Files.find_files
-
-      diff = Set.new(@__files_before_test) ^ Set.new(files_after_test)
-      expect(diff).to be_empty, <<~EOS
-        file leak detected:
```

Comment:
This is the reason: We want to detect leaked files we don't expect.

Reply:
It seems like the leaked file detection only operates in a limited scope so I guess I’m wondering how useful it really is. You can write to the wrong directory easily and this won’t catch it if the wrong directory gets cleaned up manually (1 of the 30+ that get cleaned up like that) or if it writes somewhere outside of the temporary test directory (which happens in at least one test).

It seems like it will only catch cases where you write directly to the temporary test directory (I don't know why normal code would ever attempt that) or to another directory in `HOMEBREW_PREFIX` that doesn't get cleaned up automatically.

I overlooked this as not being very useful.

Reply:
I guess my next question is how do we decide what should get cleared automatically. It feels like if we wanted to maximize the possibility of catching file leaks we would reduce the size of this list.

I'll close this PR though.

Reply:
> It seems like the leaked file detection only operates in a limited scope so I guess I’m wondering how useful it really is.

Given how much cleanup is going on between every test: I am not very convinced it's useful at all any more.

> I guess my next question is how do we decide what should get cleared automatically. It feels like if we wanted to maximize the possibility of catching file leaks we would reduce the size of this list.

Agreed but that also means more individual tests will have to do this busywork.



> I overlooked this as not being very useful.
> I'll close this PR though.

Personally: I still think this PR is worth having so feel free to reopen if you agree.